exp/services/recoverysigner/internal/db/dbmigrate/migrations/20200311000000-create-accounts.sql

```diff
@@ -0,0 +1,15 @@
+-- +migrate Up
+
+CREATE TABLE accounts (
+  id UUID NOT NULL PRIMARY KEY,
+  created_at TIMESTAMP WITH TIME ZONE NOT NULL,
```

Comment:
```suggestion
  created_at TIMESTAMP WITH TIME ZONE NOT NULL DEFAULT NOW(),
```

Since the column is for `created_at`, I usually have the db do `Now()`, which returns a timestamp with timezone, and I haven't had any problem relying on db. If we were to use `timestamp`, we would essentially cut the timezone off when we store it (if we want to use the database function and if the conversion works.)

Reply:
I also like using `DEFAULT NOW()` for the `created_at` field. If you agree with that @leighmcculloch, please remember to add it to the other tables as well:
* [identities table](https://github.com/stellar/go/pull/2386/files#diff-ac186b3068915e04924859e4ff6372b7R10).
* [auth_methods table](https://github.com/stellar/go/pull/2386/files#diff-2e653a05c60233a279a2972786c41c1fR14).

Reply:
I've used setting a default value before too, why do you guys like using it?

It definitely works, but it makes writing tests difficult.
- Instead of writing simple tests that check that a set of rows exist in the database, we have to write tests that check each value independently and do things like check that the value is within a certain duration. In some languages and testing frameworks (e.g. Ruby with Rspec) this is really easy, in Go not so much.
- Mocking time also becomes more difficult.

In the application code it's really easy to set the `created_at` with a value which is what I'm doing in the code at the moment. Do you see an issue with that?

Reply:
Hum, I was not thinking from that perspective... I agree with not including ~`DEFAULT NOW()`~.

Reply:
> we have to write tests that check each value independently and do things like check that the value is within a certain duration. In some languages and testing frameworks (e.g. Ruby with Rspec) this is really easy, in Go not so much.

How is it more difficult with Go than with Rspec? Can you elaborate? Note that we can also write a query to check whether the timestamp is within a certain range if you don't like the Go `time` package. In addition, I don't understand how it is going to make a difference in testing whether we generate a timestamp in the function or in the database. After all, we need to test if the function stores the data correctly, and the client shouldn't be concerned about whether the timestamp is generated before or during the db call, right?

I don't feel strongly about this, but atm I still prefer to have `DEFAULT NOW()` rather than writing our own logic in the application level.

Reply:
One interesting issue with setting this in the application layer is that, if we have 2+ server instances behind a load balancer, there's a chance that they will have slightly skewed clocks, which could lead to a false history of the order in which rows were created (or even you could see an entry with a higher incremental `pk` but lower `created_at` than another). 

Allowing the database to be the source of time truth in this case mitigates the issue, but I'm not sure if this is a big issue.

Reply:
I don't think that's a big issue. You can get that inconsistency between PKs and timestamps anyway because `NOW()` is the timestamp of the start of the database transaction, not the time of the insert statement, and the PK is incremented when the insert statement is executed inside the transaction not at the start of the transaction. Once an app gets to scale of multiple write db servers there's also chance with inconsistency.

This app logic does other important things with time (e.g. validate JWTs) so I'd hope we weren't allowing too great a time skew.